### PR TITLE
Support CP (no sequence packing)

### PR DESCRIPTION
**Pull request by ankisinha-nvidia**
# What does this PR do ?
Align the RL CP>1 path with the working branch and add launcher/inference fixes so interactive smoke runs can collect rollouts, surface inference progress, and complete short training iterations reliably. 

:warning: For major changes (either in lines of code or in its impact), please make sure to first share a design doc with the team. If you're unsure what's the best way to do so, contact the @mcore-oncall.

### Pre-checks

- [ ] I have added relevant unit tests
- [ ] I have added relevant functional tests
- [ ] I have added proper typing to my code [Typing guidelines](https://docs.python.org/3/library/typing.html)
- [ ] I have added relevant documentation
- [ ] I have run the [autoformatter.sh](https://github.com/NVIDIA/Megatron-LM/blob/main/tools/autoformat.sh) on my PR

### Code review

Feel free to message or comment the [@mcore-oncall](https://github.com/orgs/NVIDIA/teams/mcore-oncall) to help accelerate your merge into main. The less complex your PR is, the faster it will be approved and merged!

All PRs start as **draft**. If you open a non-draft PR, it will be automatically converted to draft.

#### Step 1: Mark PR as "Ready for Review"

1. When your PR is ready, click **Ready for Review**.
2. An oncall reviewer is auto-assigned and expert reviewers are notified based on your changes.
   - Some PRs may jump straight to step 2. This is determined by `.github/CODEOWNERS`.

:warning: Only mark as ready once merge-conflicts are resolved and the CI is passing.
Final Review might get declined if these requirements are not fulfilled.

#### Step 2: Final Review

For PRs that change `megatron/core`, once all expert reviewers have approved, the `Final Review` label is applied **automatically** and final reviewers are assigned.

For PRs outside `megatron/core`, this step is skipped.

#### Step 3: Approved

Once all required reviewers have approved, the `Approved` label is applied **automatically**.

### Merge

Any member of [mcore-engineers](https://github.com/orgs/NVIDIA/teams/mcore-engineers) will be able to merge your PR.

<details>
<summary>For MRs into `dev` branch</summary>
The proposed review process for `dev` branch is under active discussion.

MRs are mergable after one approval by either `eharper@nvidia.com` or `zijiey@nvidia.com`.
</details>

### CODE REVIEW

File: megatron/rl/rl_utils.py [hunk 1]
```diff
@@ -637,7 +635,75 @@ def selective_log_softmax(logits, index):
...
+        if tensor.shape[1] < seq_length:
+            tensor = torch.nn.functional.pad(tensor, (0, seq_length - tensor.shape[1]))
+        batch[name] = tensor
+
+    return get_batch_on_this_cp_rank(batch)
+
+
+def cp_gather_logprobs(local_logprobs):
+    """All-gather and un-zigzag CP-local logprobs back to [B, S-1]."""
+    cp_size = mpu.get_context_parallel_world_size()
```
Comment (wdykas) [inline megatron/rl/rl_utils.py]: please dont grab from global groups. you can see grabbing groups here https://github.com/ankisinha-nvidia/Megatron-LM/blob/24d79d247a683ef4648039d39c657c7257d19871/megatron/rl/rl_utils.py#L527 how it can be done

File: megatron/rl/rl_utils.py [hunk 2]
```diff
@@ -637,7 +635,75 @@ def selective_log_softmax(logits, index):
...
+    if cp_size <= 1:
+        return local_logprobs
+
+    gathered = [torch.empty_like(local_logprobs) for _ in range(cp_size)]
+    torch.distributed.all_gather(
+        gathered, local_logprobs.contiguous(), group=mpu.get_context_parallel_group()
```
Comment (wdykas) [inline megatron/rl/rl_utils.py]: same comment as above

File: megatron/rl/rl_utils.py [hunk 3]
```diff
@@ -658,12 +724,16 @@ def get_logprobs(model, tokens, position_ids, no_grad=False, sequence_packing=Fa
...
-    # When sequence_packing is disabled, construct a single-sequence default so the CUDA
-    # graph signature matches the training forward_step in train_rl.py.
-    # This is necessary because reference logprobs steps will reuse the training forward graph.
-    if packed_seq_params is None:
+    # For non-packed RL with CP>1, keep packed_seq_params=None to stay on the SBHD path so
+    # RoPE and attention use the same zigzag CP partitioning as the split batch.
+    use_cp_nonpacked_path = (
+        labels is not None
+        and not sequence_packing
+        and mpu.get_context_parallel_world_size() > 1
```
Comment (wdykas) [inline megatron/rl/rl_utils.py]: same as comments above
